src/vivarium_public_health/population/base_population.py

```diff
@@ -235,16 +235,56 @@ def _load_population_structure(self, builder: Builder) -> pd.DataFrame:
             raise ValueError(
                 f"Scaling factor must be a pandas DataFrame. Provided value: {scaling_factor}"
             )
-        scaling_factor = scaling_factor.set_index(
-            [col for col in scaling_factor.columns if col != "value"]
+        formatted_pop_structure, formatted_scaling_factor = self._format_data_inputs(
+            population_structure, scaling_factor
         )
-        population_structure = population_structure.set_index(
-            [col for col in population_structure.columns if col != "value"]
-        )
-        scaled_population_structure = (population_structure * scaling_factor).reset_index()
+
+        scaled_population_structure = (
+            formatted_pop_structure * formatted_scaling_factor
+        ).reset_index()
 
         return scaled_population_structure
 
+    def _format_data_inputs(
+        self, pop_structure: pd.DataFrame, scalar_data: pd.DataFrame
+    ) -> tuple[pd.DataFrame, pd.DataFrame]:
+        """Data cleaning function to check whether scalar_data and population structure are compatible for scaling
+        the population structure of a simulation. This method will do any necessary transformations to
+        deal with two use cases where either population structure has multiple years of data, or both data
+        inputs have multiple years of data."""
+
+        scaling_factor = scalar_data.set_index(
+            [col for col in scalar_data.columns if col != "value"],
+            drop=True,
+        )
+        population_structure = pop_structure.set_index(
+            [col for col in pop_structure.columns if col != "value"],
+            drop=True,
+        )
+        if "year_start" not in scaling_factor.index.names:
```

Comment:
so this means that the scaling is to be applied to all years?

Reply:
Yes if the year index levels aren't in scaling factor it will be able to be cast across all years in the population structure.